go/types/set_iterator.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import "github.com/attic-labs/noms/go/d"
+
+type SetIterator interface {
+	// Next returns nil, when no objects remain
+	Next() Value
+
+	// AdvanceTo(v) advances the cursor to a new position based on the following rules and returns the value at that position.
+	// * if v <= curValue, advances to the next value in the iteration.
```

Comment:
More thoughts:

The behaviour that if the current value < v then it advances is strange. If I have [1,2,3,4,5], current value is 3, and I "advance to" 1, then the iterator is now at 4. That's counterintuitive to me.

I still think this would ideally be either better named, or have a different interface, or different behavior depending on which is most sensible.

E.g. what if the comment were actually "like Next(), but returns a value > `v`", which I think is what it's doing at the moment? That makes a lot more sense to me (and the name NextFrom evokes that more than AdvanceTo, but don't feel particularly strongly about that).